workflow-examples/src/main/java/com/redhat/parodos/examples/prebuilt/migrationtoolkit/ProcessAnalysisTask.java

```diff
@@ -0,0 +1,130 @@
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.security.KeyManagementException;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+import java.util.List;
+import java.util.function.Predicate;
+
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.X509TrustManager;
+
+import com.redhat.parodos.workflow.context.WorkContextDelegate;
+import com.redhat.parodos.workflow.exception.MissingParameterException;
+import com.redhat.parodos.workflow.option.WorkFlowOption;
+import com.redhat.parodos.workflow.option.WorkFlowOptions;
+import com.redhat.parodos.workflow.task.assessment.BaseAssessmentTask;
+import com.redhat.parodos.workflows.work.DefaultWorkReport;
+import com.redhat.parodos.workflows.work.WorkContext;
+import com.redhat.parodos.workflows.work.WorkReport;
+import com.redhat.parodos.workflows.work.WorkStatus;
+import org.jsoup.Jsoup;
+import org.jsoup.select.Elements;
+
+public class ProcessAnalysisTask extends BaseAssessmentTask {
+
+	private final WorkFlowOption passOption;
+
+	private final WorkFlowOption defaultOption;
+
+	private final Predicate<MTAAnalysisReport.AnalysisIncidents> incidentsPredicate;
+
+	/**
+	 * @param passOption option for a successful report analysis
+	 * @param defaultOption default option to return
+	 * @param passCriteria the predicate to test against the report incidents from the
+	 * analysis task
+	 */
+	protected ProcessAnalysisTask(WorkFlowOption passOption, WorkFlowOption defaultOption,
+			Predicate<MTAAnalysisReport.AnalysisIncidents> passCriteria) {
+		super(List.of(passOption, defaultOption));
+		this.passOption = passOption;
+		this.defaultOption = defaultOption;
+		this.incidentsPredicate = passCriteria;
+	}
+
+	@Override
+	public WorkReport execute(WorkContext workContext) {
+		String reportURL;
+		try {
+			reportURL = getRequiredParameterValue("reportURL");
+		}
+		catch (MissingParameterException e) {
+			return new DefaultWorkReport(WorkStatus.FAILED, workContext);
+		}
+
+		// this is the "migration issues" section from report from url.It assumed to be in
+		// HTML format. In version 6.1 of MTA it can be
+		// downloaded as csv and would be simpler to process, hopefully.
+		URI uri = URI.create(reportURL).resolve("reports");
+		try {
+			var reports = downloadReport(uri);
+			Elements reportLinks = Jsoup.parse(reports).select("a[href~=report_index_]");
+			if (reportLinks.isEmpty()) {
+				return new DefaultWorkReport(WorkStatus.FAILED, workContext,
+						new Exception("failed to parse report index. Missing reports/report_index_*"));
+			}
+			String migrationIssuesReport = downloadReport(uri.resolve(reportLinks.get(0).text()));
```

Comment:
reportLinks.get(0) is "<a href="reports/report_index_demo.html" style="float: left; margin-right: 5px; margin-top: 6px;"> *** </a>"

Reply:
like mentioned on chat - I dropped the html rendering in favour of parsing the javascript data. all this code is gone now